src/main/src/test/java/org/geoserver/util/EntityResolverProviderTest.java

```diff
@@ -49,4 +53,50 @@ public void testAllowListDomains() throws Exception {
         assertTrue(allowed.containsAll(Arrays.asList(AllowListEntityResolver.OGC.split("\\|"))));
         assertTrue(allowed.contains("how2map.com"));
     }
+
+    @Test
+    public void testNotRegEx() throws Exception {
+        Set<String> allowed = EntityResolverProvider.entityResolutionAllowlist("foo*bar");
+        assertNotNull(allowed);
+    }
+
+    @Test
+    public void testEntityResolver() throws Exception {
```

Comment:
Thanks for putting in a good Test Case!  This entity stuff is obviously more complex than it seems!

1. I think the test case name should be a little more specific.
2. (optional, but appreciated) It might be very helpful to put in a java doc saying what its doing
3. (optional) I'd like to see a few more tests.  Perhaps a "good" resolution?  Perhaps one that has some "funny" URL encoding to mask the externality or get around the regex?????  I'll let you decide if they are necessary or not.